server/src/main/java/org/opensearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -342,9 +342,10 @@ long estimateRamBytesUsedForReduce(long size) {
             return Math.round(1.5d * size - size);
         }
 
-        public void consume(QuerySearchResult result, Runnable next) {
+        public void consume(QuerySearchResult result, Runnable next) throws CircuitBreakingException {
             boolean executeNextImmediately = true;
             synchronized (this) {
+                checkCircuitBreaker();
```

Comment:
While having `CircuitBreaking` check is good, the limits are generally much higher for circuit breaker as they are aggregated at node level. `SearchBackpressure/WLM` might have cancelled the search task much earlier, hence might be good idea to ensure `SearchTask` is not cancelled to enforce the `WorkloadGroup` limits

Reply:
Added changes to consider task cancellation as well